app/services/vacancy_filters.rb

```diff
@@ -57,13 +57,15 @@ def any?
 
   private
 
-  def extract_working_pattern(params)
-    params[:working_pattern] if Vacancy.working_patterns.include?(params[:working_pattern])
+  def extract_working_pattern(working_pattern)
+    working_pattern if Vacancy.working_patterns.include?(working_pattern)
   end
 
-  def extract_phases(params)
-    return if params[:phases].blank?
+  def extract_phases(phases)
+    return if phases.blank?
 
-    JSON.parse(params[:phases]).select { |phase| School.phases.include?(phase) }.presence
+    phases = JSON.parse(phases) if phases.is_a?(String)
```

Comment:
It looks like we have this ` JSON.parse ` functionality for "phases" in both the controller and this class.

What is encoding the phases as JSON?

Reply:
When we save a `Subscription`, we encode the `search_criteria` as JSON. We parse this JSON in full, meaning `phases` is now an array.

When we get a new search, we get a series of string parameters. This means that `phases` is a string representation of an array that needs parsing.

Probably we should do that parsing earlier.

Reply:
Created a card to look into this a bit more: https://trello.com/c/fPWAOZKD/887-refactor-search-criteria